### PR TITLE
New: Add rule no-triple-slash-reference

### PR DESCRIPTION
**Pull request by dannyfritz**
Rule based off of TSLint's [no-reference rule](http://palantir.github.io/tslint/rules/no-reference/)

### CODE REVIEW

File: docs/rules/no-triple-slash-reference.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Enforces reference comments are not used (no-triple-slash-reference)
+
+Reference comments, `/// <reference />`, should not be used anymore. Use `import` instead.
```
Comment (nzakas) [inline docs/rules/no-triple-slash-reference.md]: Can you explain this a bit more? As someone unfamiliar with TypeScript, I think more context would be helpful.

File: tests/lib/rules/no-triple-slash-reference.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+var ruleTester = new RuleTester();
+ruleTester.run("no-triple-slash-reference", rule, {
+
+    valid: [
+        {
```
Comment (nzakas) [inline tests/lib/rules/no-triple-slash-reference.js]: Can you add a valid test that uses comments? I'd just like to be sure the rule doesn't incorrectly warn on double-slash, for example.

File: tests/lib/rules/no-triple-slash-reference.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            code: "import Animal from \"Animal\"",
```
Comment (nzakas) [inline tests/lib/rules/no-triple-slash-reference.js]: This really isn't a useful test because the rule isn't looking for import statements.

**Comment by dannyfritz** [comment]
Requested changes have been made.

**Comment by nzakas** [comment]
Thanks!

**Pull request merged**